.github/workflows/build.yml

```diff
@@ -6,9 +6,18 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions-rs/toolchain@v1
-      with:
-        toolchain: stable
-        override: true
-    - run: cargo test --all
\ No newline at end of file
+      - uses: actions/checkout@v2
+      - name: Cache Cargo
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            ~/.cargo/bin
+            target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
Hm, not so familiar with this, but would the following be useful to include in the key:
* Compiler version, otherwise the cache could grow indefinitely in size between releases.
* `toolchain`, since stable won't benefit from a cached nightly bulid and vice-versa.

Cargo lock is also not checked in. So not sure it makes a difference here.

Reply:
Will update that shortly.

Reply:
Fixed, we are using `rustc` hash now
see https://github.com/actions-rs/toolchain#outputs